### ISSUE TITLE
request zypp(er).conf flag for --auto-agree-with-licenses

### ISSUE THREAD

**Issue by sdoconnell** [openSUSE/zypper#220]
It would be useful in some cases of automated deployments to have a flag in /etc/zypp/zypp(er).conf to automatically agree with licenses as a system-wide default, avoiding the need to specify the `--auto-agree-with-licenses` or `-l` options when calling `zypper [in|up|dup]`.

**Comment by mlandres** [comment]
Somewhat reluctant (also not sure if it has legal impacts). It should IMO be easy especially for scripted environments and automated deployments to `alias zypp='zypper --auto-agree-with-licenses'`.

**Comment by sdoconnell** [comment]
I looked into that and found that `--auto-agree-with-licenses` is a flag on various subcommands (e.g., `in`, `up`, `dup`, `patch`, etc.) and not on the root `zypper` command. Thus, you would need to create multiple aliases to accommodate that. That's kind of why I thought it might be useful to have a config option for the flag. Perhaps an option that only applies to `up`, `dup`, `patch`, or other operations on an already installed package, with `in` still requiring either the flag or a manual agreement to the license? At least in that case, you would only be dealing with a single alias for the initial install operation (`alias zinstall='zypper in --auto-agree-with-licenses`).

Though honestly from a legal perspective, I fail to see how a config flag would be any different than an admin creating an alias. The end result is the same, an implicit agreement to the software license.

**Comment by mlandres** [comment]
Indeed, I overlooked that is't not a global option. I'm sorry. You are right, as a command option it's not nice to handle. And legal also confirmed that they don't see an issue if it's not on per default.
We'll have a closer look at it...

**Comment by sdoconnell** [comment]
Cool. Thanks Michael!
